packages/nextjs/src/config/wrappers/wrapperUtils.ts

```diff
@@ -1,46 +1,58 @@
  *
- * @template T Types for `getInitialProps`, `getStaticProps`, and `getServerSideProps`
- * @param origFunction The user's exported `getInitialProps`, `getStaticProps`, or `getServerSideProps` function
- * @param context The context object passed by nextjs to the function
- * @returns The result of calling the user's function
+ * We only do the following until we move transaction creation into this function: When called, the wrapped function
+ * will also update the name of the active transaction with a parameterized route provided via the `options` argument.
  */
-export async function wrapperCore<T extends DataFetchingFunction>(options: {
-  origFunction: T['fn'];
-  context: T['context'];
-  route: string;
-  op: string;
-}): Promise<T['result']> {
-  const { origFunction, context, route, op } = options;
+export function callDataFetcherTraced<F extends (...args: any[]) => Promise<any> | any>(
+  origFunction: F,
+  origFunctionArgs: Parameters<F>,
+  options: {
+    route: string;
+    op: string;
+  },
+): ReturnType<F> {
+  const { route, op } = options;
 
   const transaction = getActiveTransaction();
 
-  if (transaction) {
-    // TODO: Make sure that the given route matches the name of the active transaction (to prevent background data
-    // fetching from switching the name to a completely other route)
-    transaction.name = route;
-    transaction.metadata.source = 'route';
-
-    // Capture the route, since pre-loading, revalidation, etc might mean that this span may happen during another
-    // route's transaction
-    const span = transaction.startChild({ op, data: { route } });
-
-    // TODO: Can't figure out how to tell TS that the types are correlated - that a `GSPropsFunction` will only get passed
-    // `GSPropsContext` and never, say, `GSSPContext`. That's what wrapping everything in objects and using the generic
-    // and pulling the types from the generic rather than specifying them directly was supposed to do, but... no luck.
-    // eslint-disable-next-line prefer-const, @typescript-eslint/no-explicit-any
-    const props = await (origFunction as any)(context);
-
-    span.finish();
-
-    return props;
+  if (!transaction) {
+    return origFunction(...origFunctionArgs);
   }
 
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  return (origFunction as any)(context);
+  // TODO: Make sure that the given route matches the name of the active transaction (to prevent background data
+  // fetching from switching the name to a completely other route) -- We'll probably switch to creating a transaction
+  // right here so making that check will probabably not even be necessary.
+  // Logic will be: If there is no active transaction, start one with correct name and source. If there is an active
+  // transaction, create a child span with correct name and source.
+  // We will probably need to put
```

Comment:
... ?

Reply:
Just some random thought I forgot to finish. I can't remember what I was about to write so I deleted it.